File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        inner = ""
        depth = 0
        
        for c in s:
            if c == '(':
                depth += 1
            if depth > 1:
                inner += c
            if c == ')':
                depth -= 1
        return inner

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        StringBuilder inner = new StringBuilder();
        int depth = 0;
        
        for (char c : s.toCharArray()) {
            if (c == '(')
                depth++;
            if (depth > 1)
                inner.append(c);
            if (c == ')')
                depth--;
        }
        return inner.toString();
    }
}